Update vcpkg-tool to 2023-02-16.

BillyONeal opened the pull request:
https://github.com/microsoft/vcpkg-tool/releases/tag/2023-02-16

BillyONeal committed:
Update vcpkg-tool to 2023-02-16.

autoantwort commented:
You probably also have to merge https://github.com/microsoft/vcpkg-tool/pull/906 to get a green ci.

dg0yt commented:
The new CI regressions are expected, due to unblocking of port builds by no longer pulling in unsupported features via unsupported features and ports. This is the work which started with #28619, lead to https://github.com/microsoft/vcpkg-tool/pull/846, but was ignored in the course of merging https://github.com/microsoft/vcpkg-tool/pull/856 which unblocked even more ports.

As a quick mitigation, I suggest to compare with the CI build before this PR, and then add expected failures to the CI baseline.

walbourn commented:
Verified this update resolves all the issues with the new **xbox** supports term.

BillyONeal committed:
[vcpkg-gfortran] Use VCPKG_POLICY_ALLOW_OBSOLETE_MSVCRT rather than VCPKG_POLICY_SKIP_DUMPBIN_CHECKS now that checking for msvc no longer requires dumpbin.

BillyONeal committed:
[blend2d] Fix usage post-build check failure and add quotes.

BillyONeal committed:
[clipper2,jxrlib] Fix usage install.

BillyONeal committed:
[qt5-x11extras] Disable on all Windows.

This never worked on Windows but was guarded before by failed dependencies on arm.

BillyONeal committed:
[boost] Add "supports": "!uwp" to boost-random, and rerun generate-ports.ps1.

https://github.com/microsoft/vcpkg/pull/29540 and https://github.com/microsoft/vcpkg/pull/29481 forgot to rerun generate-ports.ps1.

Note that the default port-version was bumped because most everything depends on boost-build that change touched.

BillyONeal committed:
[monkeys-audio] Use /Z7 and no /Gm to avoid spurious compiler crashes.

BillyONeal committed:
[openmesh] Fix usage install on case-sensitive file systems.

BillyONeal committed:
[functions-framework-cpp] Mark unsupported on uwp.

BillyONeal committed:
[openimageio] Cherry-pick arm64-windows fixes.

BillyONeal committed:
[vtk] Ensure vtk uses pegtl-2, as declared in its dependencies list, rather than pegtl, if that port happened to get installed first.

BillyONeal commented:
Saving this screenshot demonstrating why I changed vtk:

![image](https://user-images.githubusercontent.com/1544943/219846430-616a233b-c580-422b-ad81-ed52a7674424.png)

BillyONeal committed:
[harfbuzz] Not supported on UWP.

Upstream has:
```
// in buildtrees\harfbuzz\src\6.0.0-21fda9578f.clean\src\hb-ft.h
#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP)
#define generic GenericFromFreeTypeLibrary
#endif
```
resulting in
```
../src/6.0.0-21fda9578f.clean/src/hb-ft.cc(985): error C2039: 'generic': is not a member of 'FT_FaceRec_'
C:\Dev\vcpkg\installed\x64-uwp\include\freetype/freetype.h(1077): note: see declaration of 'FT_FaceRec_'
../src/6.0.0-21fda9578f.clean/src/hb-ft.cc(1010): error C2039: 'generic': is not a member of 'FT_FaceRec_'
C:\Dev\vcpkg\installed\x64-uwp\include\freetype/freetype.h(1077): note: see declaration of 'FT_FaceRec_'
```
so this problem is an upstream issue with not supporting non-desktop Windows.

BillyONeal commented:
Similarly, what changed for nghttp2:

![image](https://user-images.githubusercontent.com/1544943/219848122-9287fc20-b283-4d68-a169-0eefd174ef6d.png)

BillyONeal commented:
```
REGRESSION: blend2d:x64-windows failed with POST_BUILD_CHECKS_FAILED. If expected, add blend2d:x64-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: blend2d:x86-windows failed with POST_BUILD_CHECKS_FAILED. If expected, add blend2d:x86-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
```
Broken by https://github.com/microsoft/vcpkg-tool/pull/877, fixed by ca1cb8a3540f45a5b95dfa040e01c3922eae33f6

```
REGRESSION: boost-random:arm-uwp failed with BUILD_FAILED. If expected, add boost-random:arm-uwp=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: boost-random:x64-uwp failed with BUILD_FAILED. If expected, add boost-random:x64-uwp=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
```
Broken by https://github.com/microsoft/vcpkg-tool/pull/856, fixed by 99c4b28ed3bd8f6b3ca76de135a62dd978fd4fe3

Note also that the whole boost ecosystem got port-version bumped because https://github.com/microsoft/vcpkg/pull/29540 and https://github.com/microsoft/vcpkg/pull/29481 forgot to rerun generate-ports.ps1.

```
REGRESSION: clipper2:arm-uwp failed with POST_BUILD_CHECKS_FAILED. If expected, add clipper2:arm-uwp=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: clipper2:arm64-windows failed with POST_BUILD_CHECKS_FAILED. If expected, add clipper2:arm64-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: clipper2:x64-linux failed with POST_BUILD_CHECKS_FAILED. If expected, add clipper2:x64-linux=fail to /agent/_work/1/s/scripts/azure-pipelines/../ci.baseline.txt.
REGRESSION: clipper2:x64-uwp failed with POST_BUILD_CHECKS_FAILED. If expected, add clipper2:x64-uwp=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: clipper2:x64-windows failed with POST_BUILD_CHECKS_FAILED. If expected, add clipper2:x64-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: clipper2:x64-windows-static failed with POST_BUILD_CHECKS_FAILED. If expected, add clipper2:x64-windows-static=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: clipper2:x64-windows-static-md failed with POST_BUILD_CHECKS_FAILED. If expected, add clipper2:x64-windows-static-md=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: clipper2:x86-windows failed with POST_BUILD_CHECKS_FAILED. If expected, add clipper2:x86-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: jxrlib:arm-uwp failed with POST_BUILD_CHECKS_FAILED. If expected, add jxrlib:arm-uwp=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: jxrlib:arm64-windows failed with POST_BUILD_CHECKS_FAILED. If expected, add jxrlib:arm64-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: jxrlib:x64-linux failed with POST_BUILD_CHECKS_FAILED. If expected, add jxrlib:x64-linux=fail to /agent/_work/1/s/scripts/azure-pipelines/../ci.baseline.txt.
REGRESSION: jxrlib:x64-uwp failed with POST_BUILD_CHECKS_FAILED. If expected, add jxrlib:x64-uwp=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: jxrlib:x64-windows failed with POST_BUILD_CHECKS_FAILED. If expected, add jxrlib:x64-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: jxrlib:x64-windows-static failed with POST_BUILD_CHECKS_FAILED. If expected, add jxrlib:x64-windows-static=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: jxrlib:x64-windows-static-md failed with POST_BUILD_CHECKS_FAILED. If expected, add jxrlib:x64-windows-static-md=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: jxrlib:x86-windows failed with POST_BUILD_CHECKS_FAILED. If expected, add jxrlib:x86-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
```
Broken by https://github.com/microsoft/vcpkg-tool/pull/877, fixed by 88775f549a891afaebadf5c5b711a954effa6c9f

```
REGRESSION: vcpkg-gfortran:x64-uwp failed with POST_BUILD_CHECKS_FAILED. If expected, add vcpkg-gfortran:x64-uwp=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: vcpkg-gfortran:x64-windows failed with POST_BUILD_CHECKS_FAILED. If expected, add vcpkg-gfortran:x64-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: vcpkg-gfortran:x64-windows-static failed with POST_BUILD_CHECKS_FAILED. If expected, add vcpkg-gfortran:x64-windows-static=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: vcpkg-gfortran:x64-windows-static-md failed with POST_BUILD_CHECKS_FAILED. If expected, add vcpkg-gfortran:x64-windows-static-md=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: vcpkg-gfortran:x86-windows failed with POST_BUILD_CHECKS_FAILED. If expected, add vcpkg-gfortran:x86-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
```
Broken by https://github.com/microsoft/vcpkg-tool/pull/834, fixed by be95ce72b161093f9f3d2847c2792229fbf991a3

```
REGRESSION: monkeys-audio:x86-windows failed with BUILD_FAILED. If expected, add monkeys-audio:x86-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
```
Error is `fatal error C1090: PDB API call failed, error code '23'`, a sign that mspdbsrv failed. This is one of the many reasons we dislike `/Zi`. 64bd5dad0e94c5fffb85b81fe79c574c790ca761 patches that out.

```
REGRESSION: harfbuzz:arm-uwp failed with BUILD_FAILED. If expected, add harfbuzz:arm-uwp=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: harfbuzz:x64-uwp failed with BUILD_FAILED. If expected, add harfbuzz:x64-uwp=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
```
Broken by https://github.com/microsoft/vcpkg-tool/pull/856 , previously this was a cascaded failure. Upstream has:
```
// in buildtrees\harfbuzz\src\6.0.0-21fda9578f.clean\src\hb-ft.h
#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP)
#define generic GenericFromFreeTypeLibrary
#endif
```
resulting in
```
../src/6.0.0-21fda9578f.clean/src/hb-ft.cc(985): error C2039: 'generic': is not a member of 'FT_FaceRec_'
C:\Dev\vcpkg\installed\x64-uwp\include\freetype/freetype.h(1077): note: see declaration of 'FT_FaceRec_'
../src/6.0.0-21fda9578f.clean/src/hb-ft.cc(1010): error C2039: 'generic': is not a member of 'FT_FaceRec_'
C:\Dev\vcpkg\installed\x64-uwp\include\freetype/freetype.h(1077): note: see declaration of 'FT_FaceRec_'
```
so this problem is an upstream issue with not supporting non-desktop Windows. Marked as unsupported in 04e8c539e848d19ae3d70b7a0445a9b031625be3


```
REGRESSION: nghttp2:arm-uwp failed with BUILD_FAILED. If expected, add nghttp2:arm-uwp=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: nghttp2:x64-uwp failed with BUILD_FAILED. If expected, add nghttp2:x64-uwp=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
```
Probably broken by https://github.com/microsoft/vcpkg-tool/pull/856, this port is full of "optional detection" stuff which disabled features before because it happened to install early, but now it happened to install much later. There are significant differences in what it reports between the "working" and "broken" states; I'm not sure how to fix this one yet.


```
# REGRESSION: functions-framework-cpp:arm-uwp failed with BUILD_FAILED. If expected, add functions-framework-cpp:arm-uwp=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
```
Broken by https://github.com/microsoft/vcpkg-tool/pull/856, needs VerifyVersionInfo not present on UWP. See also https://github.com/GoogleCloudPlatform/functions-framework-cpp/issues/207. Fixed by 45223aeca00e157af010498a32f962f1949ac37d

```
# REGRESSION: openimageio:arm64-windows failed with BUILD_FAILED. If expected, add openimageio:arm64-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
```
Broken by https://github.com/microsoft/vcpkg-tool/pull/856, simd.h missing msvc-arm64 support, Fixed by cherry picking upstream change https://github.com/OpenImageIO/oiio/commit/1aa2cf6edf9f2bd57a21998fceeb095cbe1f2b84 in 4fa6f5c5d279c26590c0cfc2da1c2b1332690b94

```
REGRESSION: openmesh:x64-linux failed with POST_BUILD_CHECKS_FAILED. If expected, add openmesh:x64-linux=fail to /agent/_work/1/s/scripts/azure-pipelines/../ci.baseline.txt.
```
Broken by https://github.com/microsoft/vcpkg-tool/pull/877, fixed by 71df8be5d5389b36b8ab6a6fa081fbc4088ad66f

```
REGRESSION: vtk:x86-windows failed with BUILD_FAILED. If expected, add vtk:x86-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
```
Probably broken by https://github.com/microsoft/vcpkg-tool/pull/856 changing the installation order and letting pegtl install before vtk, exposing port bugs in vtk. See screenshot in previous comment. Fixed by forcing vtk to use pegtl-2 by adding that to the path used by the include statement in 6b9ac36fcbcad8b2194277508ff2338b57055369


Drive by 9cd208792c4577353b044e1b542c65be8f9518dc , I didn't realize qt5-x11extras was in ci.baseline.txt already :)

```
REGRESSION: cmake-user:x64-windows cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: cmake-user:x64-windows-static cascaded, but it is required to pass. (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: cmake-user:x64-windows-static-md cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: cmake-user:x86-windows cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-mathgl:x64-windows cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-mathgl:x64-windows-static cascaded, but it is required to pass. (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-mathgl:x64-windows-static-md cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-mathgl:x86-windows cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-opencv:x64-windows cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-opencv:x64-windows-static cascaded, but it is required to pass. (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-opencv:x64-windows-static-md cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-opencv:x86-windows cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-openimageio:x64-windows cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-openimageio:x64-windows-static cascaded, but it is required to pass. (C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-openimageio:x64-windows-static-md cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-openimageio:x86-windows cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
REGRESSION: vcpkg-ci-paraview:x86-windows cascaded, but it is required to pass. (C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt).
```
Pending.

BillyONeal committed:
monkeys-audio version DB

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into tool-release-2023-02-15

# Conflicts:
#	versions/m-/monkeys-audio.json

Neumann-A commented:
> Probably broken by https://github.com/microsoft/vcpkg-tool/pull/856 changing the installation order and letting pegtl install before vtk, exposing port bugs in vtk. See screenshot in previous comment. Fixed by forcing vtk to use pegtl-2 by adding that to the path used by the include statement in https://github.com/microsoft/vcpkg/commit/6b9ac36fcbcad8b2194277508ff2338b57055369

hmm alternatively force c++17 on vtk and update the pegtl dep? (hmm theoretically the find module should already do that.)

BillyONeal commented:
> hmm alternatively force c++17 on vtk and update the pegtl dep? (hmm theoretically the find module should already do that.)

vtk says it wants pegtl-2, based on its vcpkg.json. It is getting pegtl (not -2) if that happens to be installed, because pegtl-2 installs to  
`pegtl-2:x86-windows:/include/pegtl-2/tao/pegtl.hpp`  
and pegtl (non -2) installs to  
`pegtl:x86-windows:/include/tao/pegtl.hpp`  
and some other unrelated lib put the root of the include directory on the include path earlier.

I don't know enough about vtk to support or oppose changing the dependency from `pegtl` from `pegtl-2`, but since this is not intending to give a different answer I don't want to make that change.

BillyONeal committed:
[vcpkg-gfortran] Also not supported on UWP.

```
warning: The App Container bit must be set for Windows Store apps. The following DLLs do not have the App Container bit set:

    D:\packages\vcpkg-gfortran_x64-uwp\debug\bin\libgcc_s_seh-1.dll
    D:\packages\vcpkg-gfortran_x64-uwp\debug\bin\libgfortran-5.dll
    D:\packages\vcpkg-gfortran_x64-uwp\debug\bin\libquadmath-0.dll
    D:\packages\vcpkg-gfortran_x64-uwp\debug\bin\libwinpthread-1.dll
    D:\packages\vcpkg-gfortran_x64-uwp\bin\libgcc_s_seh-1.dll
    D:\packages\vcpkg-gfortran_x64-uwp\bin\libgfortran-5.dll
    D:\packages\vcpkg-gfortran_x64-uwp\bin\libquadmath-0.dll
    D:\packages\vcpkg-gfortran_x64-uwp\bin\libwinpthread-1.dll
```

This previously succeeded in CI but the resulting program could not be distributed.

BillyONeal commented:
[vcpkg-gfortran] Also not supported on UWP.

```
warning: The App Container bit must be set for Windows Store apps. The following DLLs do not have the App Container bit set:

    D:\packages\vcpkg-gfortran_x64-uwp\debug\bin\libgcc_s_seh-1.dll
    D:\packages\vcpkg-gfortran_x64-uwp\debug\bin\libgfortran-5.dll
    D:\packages\vcpkg-gfortran_x64-uwp\debug\bin\libquadmath-0.dll
    D:\packages\vcpkg-gfortran_x64-uwp\debug\bin\libwinpthread-1.dll
    D:\packages\vcpkg-gfortran_x64-uwp\bin\libgcc_s_seh-1.dll
    D:\packages\vcpkg-gfortran_x64-uwp\bin\libgfortran-5.dll
    D:\packages\vcpkg-gfortran_x64-uwp\bin\libquadmath-0.dll
    D:\packages\vcpkg-gfortran_x64-uwp\bin\libwinpthread-1.dll
```

This previously succeeded in CI but the resulting program could not be distributed because it would fail the WACK. (And we could not detect the problem before because it passed skip-dumpbin-checks)

dg0yt commented on ports/blend2d/usage_static.txt:
```diff
@@ -0,0 +1,6 @@
+    find_package(blend2d CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE blend2d::blend2d)
+
+Also, define BL_STATIC before any blend2d includes.
```
Shouldn't this be burned in in some way?

BillyONeal committed:
[liblsl] Not supported on UWP.

```
D:\installed\x64-uwp\include\asio/detail/impl/win_iocp_io_context.ipp(535): error C2039: 'VerSetConditionMask': is not a member of '`global namespace''
D:\installed\x64-uwp\include\asio/detail/impl/win_iocp_io_context.ipp(535): error C3861: 'VerSetConditionMask': identifier not found
D:\installed\x64-uwp\include\asio/detail/impl/win_iocp_io_context.ipp(538): error C2039: 'VerifyVersionInfo': is not a member of '`global namespace''
D:\installed\x64-uwp\include\asio/detail/impl/win_iocp_io_context.ipp(538): error C3861: 'VerifyVersionInfo': identifier not found
```

Neumann-A commented:
> [vcpkg-gfortran] Also not supported on UWP.

Be aware that we had the discussion somewhere already.

BillyONeal commented:
> Be aware that we had the discussion somewhere already.

I don't necessarily want to relitigate something that has been litigated before but I see little reason in claiming to people that something is available only for it to blow up in their face when they try to ship. Do you have a link to the previous discussion which may have landed on mitigations?

dg0yt commented on ports/monkeys-audio/vcpkg.json:
```diff
@@ -1,6 +1,7 @@
 {
   "name": "monkeys-audio",
   "version-string": "9.04",
+  "port-version": 1,
```
Attention, target of another PR: #29709.

BillyONeal replied:
Merged that one and merged this with master. Thanks!

Neumann-A commented:
> I don't necessarily want to relitigate something that has been litigated before but I see little reason in claiming to people that something is available only for it to blow up in their face when they try to ship. Do you have a link to the previous discussion which may have landed on mitigations?

https://github.com/microsoft/vcpkg/pull/19413#pullrequestreview-956898159

seems like the result was that uwp is already broken but it was never marked as broken.

BillyONeal commented:
> seems like the result was that uwp is already broken but it was never marked as broken.

Yeah there I said

> Given that UWP is already broken for gfortran et al. here, don't worry about UWP interaction in whatever the solution is.

so it seems like this is just documenting that it's broken rather than actually breaking anything.

Neumann-A commented:
> so it seems like this is just documenting that it's broken rather than actually breaking anything.

That is because the lapack dependency on uwp is fulfilled via clapack.

BillyONeal committed:
[lapack-reference] Uses msvcrt.

BillyONeal commented:
[lapack-reference]:
warning: Detected outdated dynamic CRT in the following files:
    D:\packages\lapack-reference_x86-windows\debug\bin\liblapack.dll:msvcrt.dll
    D:\packages\lapack-reference_x86-windows\bin\liblapack.dll:msvcrt.dll
To inspect the dll files, use:
    dumpbin.exe /dependents mylibfile.dll

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into tool-release-2023-02-15

# Conflicts:
#	versions/m-/monkeys-audio.json

BillyONeal commented:
> Hm I don't understand why it is necessary to bump the port version of every boost port.

https://github.com/microsoft/vcpkg/pull/29540 and https://github.com/microsoft/vcpkg/pull/29481 touch `boost-build` that the entire Boost word depends on, but forgot to rerun `generate-ports.ps1`. Really, only those ports which transitively depend on boost-random should need to be affected here.

BillyONeal committed:
Try to kick the GitHub Actions bot.

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into tool-release-2023-02-15

# Conflicts:
#	ports/openimageio/portfile.cmake
#	ports/vtk/pegtl.patch
#	versions/m-/monkeys-audio.json
#	versions/o-/openimageio.json
#	versions/v-/vtk.json

BillyONeal commented:
All VTK changes removed due to https://github.com/microsoft/vcpkg/commit/8c0fe4a71d074a2682c1984da655fe547a4fdf21 landing.

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into tool-release-2023-02-15

BillyONeal committed:
[monkeys-audio] Merge.
